jerry-core/ecma/base/ecma-alloc.c

```diff
@@ -80,12 +80,37 @@ JERRY_STATIC_ASSERT (sizeof (ecma_extended_object_t) - sizeof (ecma_object_t) <=
   ALLOC (ecma_type) \
   DEALLOC (ecma_type)
 
-DECLARE_ROUTINES_FOR (object)
 DECLARE_ROUTINES_FOR (number)
 DECLARE_ROUTINES_FOR (collection_header)
 DECLARE_ROUTINES_FOR (collection_chunk)
-DECLARE_ROUTINES_FOR (string)
-DECLARE_ROUTINES_FOR (getter_setter_pointers)
+
+/**
+ * Allocate memory for ecma-object
+ *
+ * @return pointer to allocated memory
+ */
+inline ecma_object_t * __attr_always_inline___
+ecma_alloc_object (void)
```

Comment:
to reduce copy-paste, couldn't we have templates for stat-collecting alloc/free pairs?

```c
#ifdef JMEM_STATS
#define STAT_ALLOCATE_BYTES(ecma_type) \
  jmem_stats_allocate_ ## ecma_type ## _bytes (sizeof (ecma_ ## ecma_type ## _t));
#define STAT_FREE_BYTES(ecma_type) \
  jmem_stats_free_ ## ecma_type ## _bytes (sizeof (ecma_ ## ecma_type ## _t));
#else /* !JMEM_STATS */
#define STAT_ALLOCATE_BYTES(ecma_type) 
#define STAT_FREE_BYTES(ecma_type) 
#endif /* JMEM_STATS */

/**
 * Template of an allocation routine that collects statistics
 */
#define STAT_ALLOC(ecma_type) \
  ecma_ ## ecma_type ## _t * \
  ecma_alloc_ ## ecma_type (void) \
  { \
    STAT_ALLOCATE_BYTES (ecma_type) \
    return (ecma_ ## ecma_type ## _t *) jmem_pools_alloc (sizeof (ecma_ ## ecma_type ## _t)); \
  }

/**
 * Deallocation routine template that collects statistics
 */
#define STAT_DEALLOC(ecma_type) \
  void \
  ecma_dealloc_ ## ecma_type (ecma_ ## ecma_type ## _t *ecma_type ## _p) \
  { \
    STAT_FREE_BYTES (ecma_type) \
    jmem_pools_free ((uint8_t *) ecma_type ## _p, sizeof (ecma_ ## ecma_type ## _t)); \
  }

/**
 * Declaration of alloc/free routines that collect statistics for specified ecma-type.
 */
#define DECLARE_STAT_ROUTINES_FOR(ecma_type) \
  STAT_ALLOC (ecma_type) \
  STAT_DEALLOC (ecma_type)
```